oncogene.js

```diff
@@ -22,7 +22,14 @@ class Oncogene {
     makeStep(stepSize) {
         this.stepInx = Math.max(0, this.stepInx + stepSize)
 
-        if (this.stepInx >= this.steps.length) return this.renderResult()
+        if (this.stepInx >= this.steps.length) {
+            return this.renderResult()
+        }
+
+        const step = this.getStep();
+        if (this.constructor.isFunction(step.showIf) && !step.showIf(this.config)) {
+           return this.nextStep();
```

Comment:
`this.nextStep()` is an alias for `this.makeStep(1)`. Oncogene's API allows you to call `this.makeStep(-1)` or `this.prevStep()`.

If you want to go back to the previous step, and the previous step shouldn't be showed, you should go back, not forward.